work_queue/src/work_queue.c

```diff
@@ -4137,6 +4191,26 @@ static int send_one_task( struct work_queue *q )
 	return 0;
 }
 
+static void check_all_tasks(struct work_queue *q)
+{
+	struct work_queue_task *t;
+
+	list_first_item(q->ready_list);
+	
+
+	while( (t = list_next_item(q->ready_list))){
+		// check each task against the queue of connected workers
+		if (!check_task_fit_in_connected_workers(q, t)){
```

Comment:
This could be a very long debug output. We can just print a handful of them and warn than there are others, or we can put them in a list and summarize them somehow.

For the summary, one way I was thinking is to get the maximum per resource of cores, memory, disk, and gpus of all the workers, and the max of cores, memory, disk, and gpus of the tasks that don't fit. Then, per resource, we check how many tasks don't fit in the maximum and print something like:
```
43 waiting tasks need more than 2 cores (the maximum available). The largest task needs 4 cores.
10 waiting tasks need more than 1024 MB of memory (the maximum available). The largest task needs 2000 MB.
```
It may be that all tasks do fit some resource. E.g., 0 gpus needed, 0 gpus offered by workers. If no waiting task goes above the maximum available, then we print nothing.

What do you think?

Reply:
Yeah I really like this actually I thing the first portion would be quite easy to implement, as I can just count up the number of tasks that. I think I can use a sort of bit mask to determine which resources a task check fails. 

I think I would reasonably be able to scan the largest task as well.

Reply:
using work _queue_get_stats to get the maximum stats for the workers

Reply:
Please take a look at how I tried to get the max stats for the tasks